### PR TITLE
cmocka updates for host tests and GCC xtensa

### PR DESCRIPTION
**Pull request by lgirdwood**
This PR allows the tests to be run on the host (i.e. build all mocks as local command line tests) and also to build on xtensa GCC. The xtensa GCC support is build only, as there require more work to run on qemu.

### CODE REVIEW

File: test/cmocka/cmocka-host-gcc-toolchain.cmake [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+set(CMAKE_ASM_COMPILER_ID GNU)
+set(CMAKE_C_COMPILER_ID GNU)
+
+set(CMAKE_C_COMPILER gcc)
+
+find_program(CMAKE_LD NAMES "$ld" PATHS ENV PATH NO_DEFAULT_PATH)
+find_program(CMAKE_AR NAMES "$ar" PATHS ENV PATH NO_DEFAULT_PATH)
+find_program(CMAKE_OBJCOPY NAMES "objcopy" PATHS ENV PATH NO_DEFAULT_PATH)
+find_program(CMAKE_OBJDUMP NAMES "objdump" PATHS ENV PATH NO_DEFAULT_PATH)
```
Comment (lyakh) [inline test/cmocka/cmocka-host-gcc-toolchain.cmake:19]: this might well be correct, just a bit strange why we use `NO_DEFAULT_PATH`?

**Review by singalsu — APPROVED** [review]
Looks good to me, I didn't run this myself yet.

File: test/cmocka/src/common_mocks.c [hunk 2]
```diff
@@ -337,3 +337,110 @@ uint8_t * WEAK get_library_mailbox(void)
...
+}
+
+double __divdf3(double a, double b)
+{
+	return a / b;
+}
+
+int WEAK __ledf2(double a, double b)
+{
+	return a <= b ? 0 : 1;
```
Comment (lyakh) [inline test/cmocka/src/common_mocks.c:439]: `return b < a;`

File: test/cmocka/src/common_mocks.c [hunk 3]
```diff
@@ -337,3 +337,110 @@ uint8_t * WEAK get_library_mailbox(void)
...
+}
+
+int WEAK __eqdf2(double a, double b)
+{
+	return a == b ? 0 : 1;
```
Comment (lyakh) [inline test/cmocka/src/common_mocks.c:444]: `return a != b;` but in fact this is really just a dummy, yes? We don't care if it works? If we do maybe we have to do something like `return fabs(a - b) < 0.0001;` or similar

**Review by lyakh — APPROVED** [review]

File: test/cmocka/src/common_mocks.c (hunk 2 above)
Comment (lgirdwood) [reply test/cmocka/src/common_mocks.c:439]: See https://gcc.gnu.org/onlinedocs/gccint/Soft-float-library-routines.html

```
Runtime Function: int __lesf2 (float a, float b)
Runtime Function: int __ledf2 (double a, double b)
Runtime Function: int __letf2 (long double a, long double b)
These functions return a value less than or equal to zero if neither argument is NaN, and a is less than or equal to b.
```

File: test/cmocka/src/common_mocks.c (hunk 3 above)
Comment (lgirdwood) [reply test/cmocka/src/common_mocks.c:444]:
```
Runtime Function: int __eqsf2 (float a, float b)
Runtime Function: int __eqdf2 (double a, double b)
Runtime Function: int __eqtf2 (long double a, long double b)
These functions return zero if neither argument is NaN, and a and b are equal.
```

Comment (lgirdwood) [reply test/cmocka/src/common_mocks.c:444]: Yes, they are fake functions to resolve linkage on xtensa GCC (until we can link with real libraries).

File: test/cmocka/cmocka-host-gcc-toolchain.cmake (hunk 1 above)
Comment (lgirdwood) [reply test/cmocka/cmocka-host-gcc-toolchain.cmake:19]: This aligns with the current code for XCC and can be used to use to specify specific CC versions (which is already supported as one of the cmocka cmake options). We can always revisits if this is a problem later, but it seems OK atm.

**Pull request merged**